lib/cretonne/src/ir/dfg.rs

```diff
@@ -344,14 +344,36 @@ impl DataFlowGraph {
         DisplayInst(self, inst)
     }
 
-    /// Get the value arguments on `inst` as a slice.
+    /// Get all value arguments on `inst` as a slice.
     pub fn inst_args(&self, inst: Inst) -> &[Value] {
-        self.insts[inst].arguments(&self.value_lists)
+        let (fixed, variable) = self.insts[inst].arguments(&self.value_lists);
+        if fixed.len() != 0 { fixed } else { variable }
     }
 
-    /// Get the value arguments on `inst` as a mutable slice.
+    /// Get all value arguments on `inst` as a mutable slice.
     pub fn inst_args_mut(&mut self, inst: Inst) -> &mut [Value] {
-        self.insts[inst].arguments_mut(&mut self.value_lists)
+        let (fixed, variable) = self.insts[inst].arguments_mut(&mut self.value_lists);
+        if fixed.len() != 0 { fixed } else { variable }
+    }
+
+    /// Get the fixed value arguments on `inst` as a slice.
+    pub fn inst_fixed_args(&self, inst: Inst) -> &[Value] {
```

Comment:
I think it is OK to add this method, but don't change the type returned from `InstructionData::arguments()`. I changed this method recently so it always returns a single slice, and that really simplifies those cases that don't care about the difference between fixed and variable arguments.

This method could simply return `self.inst_args(inst)[0..fixed_values]`.